libs/cdk/internals/scheduler/src/lib/scheduler.ts

```diff
@@ -456,20 +462,42 @@ const performWorkUntilDeadline = () => {
   needsPaint = false;
 };
 
-const channel: MessageChannel =
-  ɵglobal.MessageChannel && new ɵglobal.MessageChannel();
-const port = channel && channel.port2;
+let schedulePerformWorkUntilDeadline;
+if (typeof localSetImmediate === 'function') {
+  // Node.js and old IE.
+  // There's a few reasons for why we prefer setImmediate.
+  //
+  // Unlike MessageChannel, it doesn't prevent a Node.js process from exiting.
+  // (Even though this is a DOM fork of the Scheduler, you could get here
+  // with a mix of Node.js 15+, which has a MessageChannel, and jsdom.)
+  // https://github.com/facebook/react/issues/20756
+  //
+  // But also, it runs earlier which is the semantic we want.
+  // If other browsers ever implement it, it's better to use it.
+  // Although both of these would be inferior to native scheduling.
+  schedulePerformWorkUntilDeadline = () => {
+    localSetImmediate(performWorkUntilDeadline);
+  };
+} else if (typeof ɵglobal.MessageChannel !== 'undefined') {
+  const channel: MessageChannel = new ɵglobal.MessageChannel();
+  const port = channel.port2;
 
-if (channel) {
   channel.port1.onmessage = performWorkUntilDeadline;
+  schedulePerformWorkUntilDeadline = () => {
+    port.postMessage(null);
+  };
+} else {
+  // We should only fallback here in non-browser environments.
+  schedulePerformWorkUntilDeadline = () => {
+    localSetTimeout(performWorkUntilDeadline, 0);
+  };
 }
```

Comment:
@hoebbelsB I think we should unit test this, to fix the Scheduler tests I just skipped the `setImmediate` usage by setting it to `undefined` globally, but we should test the Scheduler uses the correct scheduling mechanism.

1. `setImmediate` if defined (for Node)
2. `MessageChannel` if defined
3. `setTimeout` fallback